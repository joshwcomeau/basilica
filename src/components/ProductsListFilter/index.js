// eslint-disable-next-line no-unused-vars
import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';

import { capitalize } from '../../utils/misc.utils';
import './index.scss';


const ProductsListFilter = ({ filters, activeFilter, onClickFilter }) => {
  const classes = classNames(['products-list-filter']);

  const filterElements = filters.map((filter) => {
    const filterClasses = classNames({
      'is-active': filter === activeFilter,
    });

    return (
      <li className={filterClasses} key={filter}>
        {/* Adding a <button> to be a11y-compliant */}
        <button onClick={() => onClickFilter({ filter })}>
          {capitalize(filter)}
        </button>
      </li>
    );
  });

  return (
    <div className={classes}>
      <ul>
        {filterElements}
      </ul>
    </div>
  );
};

ProductsListFilter.propTypes = {
  filters: PropTypes.arrayOf(PropTypes.string),
  activeFilter: PropTypes.string,
  onClickFilter: PropTypes.func,
};

ProductsListFilter.defaultProps = {

};

export default ProductsListFilter;
