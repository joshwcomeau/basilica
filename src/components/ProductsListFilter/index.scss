@import '../../variables';

.products-list-filter {
  height: 64px;
  background: $white;
  text-align: center;

  li {
    position: relative;
    line-height: 64px;
    display: inline-block;
    margin: 0 10px;

    button {
      border: none;
      background: transparent;
      height: 64px;
      padding: 0 90px;
      font-size: 15px;
      font-weight: bold;
      text-transform: uppercase;
      outline: none;
      cursor: pointer;
    }

    &.is-active:after {
      content: '';
      position: absolute;
      left: 0;
      right: 0;
      bottom: 0;
      height: 3px;
      background: $blue;
    }
  }

  @media screen and (max-width: $mq-tablet) {

  }

  @media screen and (max-width: $mq-mobile) {

  }
}
