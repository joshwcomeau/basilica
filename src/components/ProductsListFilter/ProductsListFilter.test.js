/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import ProductsListFilter from './index';

describe('ProductsListFilter', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ProductsListFilter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
