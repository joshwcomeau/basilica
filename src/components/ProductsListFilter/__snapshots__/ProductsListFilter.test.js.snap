exports[`ProductsListFilter renders correctly 1`] = `
<div
  className="products-list-filter">
  <ul>
    <li
      className="is-active">
      Print
    </li>
    <li>
      Original
    </li>
    <li>
      All
    </li>
  </ul>
</div>
`;
